lib/install/validate/storage.go

```diff
@@ -39,29 +39,39 @@ func (v *Validator) storage(ctx context.Context, input *data.Data, conf *config.
 		v.NoteIssue(err)
 		return
 	}
+	if err != nil {
+		v.NoteIssue(err)
+	}
 
 	// provide a default path if only a DS name is provided
 	if imageDSpath.Path == "" {
 		imageDSpath.Path = input.DisplayName
 	}
 
-	v.NoteIssue(err)
 	if ds != nil {
 		v.SetDatastore(ds, imageDSpath)
 		conf.AddImageStore(imageDSpath)
 	}
 
+	if !v.targetDatastoreIsWriteable(ctx, ds) {
+		return
+	}
+
 	if conf.VolumeLocations == nil {
 		conf.VolumeLocations = make(map[string]*url.URL)
 	}
 
 	// TODO: add volume locations
 	for label, volDSpath := range input.VolumeLocations {
-		dsURL, _, err := v.DatastoreHelper(ctx, volDSpath, label, "--volume-store")
+		dsURL, vds, err := v.DatastoreHelper(ctx, volDSpath, label, "--volume-store")
```

Comment:
We check to see if at least one clustered host is connected to each datastore. So if hosts `A` and `B` have datastores `x` and `y`, and `A` and `B` are in a cluster, and `x` is connected to `A` and `y` is connected to `B`, we'll find `y` in the list of both `B.AttachedClusterHosts` and in `Cluster.Hosts` (so we can write, and helper returns `true`) and `x` will be in `A.AttachedClusterHosts` and `Cluster.Hosts` so again helper returns `true` because we can write.